### PR TITLE
Moved variables into scope, removed unused variables and fixed variables

### PR DESCRIPTION
**Pull request by Seadragon91**
Playing around with static analyzers :)
* Moved variables into scope
* Removed unused variables
* Fixed use of wrong variable

### CODE REVIEW

File: src/PolarSSL++/BlockingSslClientSocket.cpp [hunk 1]
```diff
@@ -189,7 +189,7 @@ bool cBlockingSslClientSocket::Send(const void * a_Data, size_t a_NumBytes)
 	size_t NumBytes = a_NumBytes;
 	for (;;)
 	{
-		int res = m_Ssl.WritePlain(a_Data, a_NumBytes);
+		int res = m_Ssl.WritePlain(Data, a_NumBytes);
```
Comment (Seadragon91) [inline src/PolarSSL++/BlockingSslClientSocket.cpp]: Fixed use of wrong variable. The method `Send` is not used, but better fix that.

File: src/Entities/Entity.cpp [hunk 2]
```diff
@@ -513,7 +513,7 @@ bool cEntity::DoTakeDamage(TakeDamageInfo & a_TDI)
 			RemovedDamage += CeilC(EPFProtection * 0.04 * a_TDI.FinalDamage);
 		}
 
-		if ((a_TDI.DamageType == dtFalling) || (a_TDI.DamageType == dtFall) || (a_TDI.DamageType == dtEnderPearl))
+		if ((a_TDI.DamageType == dtFalling) || (a_TDI.DamageType == dtEnderPearl))
```
Comment (Seadragon91) [inline src/Entities/Entity.cpp]: dtFalling and dtFall have the same int value. No need to check twice :)

File: src/Blocks/BlockCrops.h [hunk 3]
```diff
@@ -75,14 +75,7 @@ class cBlockCropsHandler :
...
-		NIBBLETYPE SkyLight = a_Chunk.GetSkyLight  (a_RelX, a_RelY, a_RelZ);
-
-		if (SkyLight > Light)
-		{
-			Light = SkyLight;
-		}
+		NIBBLETYPE Meta = a_Chunk.GetMeta(a_RelX, a_RelY, a_RelZ);
 
 		// Check to see if the plant can grow
 		auto Action = CanGrow(a_Chunk, a_RelX, a_RelY, a_RelZ);
```
Comment (Seadragon91) [inline src/Blocks/BlockCrops.h]: No need for the light variables, is checked in `CanGrow`.

File: src/BlockEntities/BrewingstandEntity.cpp [hunk 4]
```diff
@@ -92,10 +92,10 @@ bool cBrewingstandEntity::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		return false;
 	}
 
-	const cBrewingRecipes::cRecipe * Recipe = nullptr;
 	// The necessary brewing time, has been reached
 	if (m_TimeBrewed >= m_NeedBrewingTime)
 	{
+		const cBrewingRecipes::cRecipe * Recipe = nullptr;
```
Comment (Seadragon91) [inline src/BlockEntities/BrewingstandEntity.cpp]: This one was my fault...

**Comment by Seadragon91** [comment]
Question about the `Rnd` [variables](https://github.com/cuberite/cuberite/blob/master/src/Generating/StructGen.cpp#L333#L335)

The variables `Rnd` aren't used in the for loop, could it be possible that `Rnd` should be [`BubbleRnd`](https://github.com/cuberite/cuberite/blob/master/src/Generating/StructGen.cpp#L329)?

File: src/PolarSSL++/BlockingSslClientSocket.cpp (hunk 1 above)
Comment (madmaxoft) [inline src/PolarSSL++/BlockingSslClientSocket.cpp]: Actually this is wrong. `Data` is the correct variable to use - the `WritePlain` method could consume less than `a_NumBytes` bytes, in which case the loop continues, but only the remaining portion of the data is to be sent (line 202).

**Comment by madmaxoft** [comment]
Good catch, it should indeed be `BubbleRnd`

**Comment by Seadragon91** [comment]
Squashed the commits, Anyone against a merge?

The static analyzer I used, also reports the (possible) use of uninitialized variables, should I open a own issue for that?

**Comment by madmaxoft** [comment]
Looks good to me.

**Comment by SafwatHalaby** [comment]
Out of curiosity, which static analyzer did you use?

**Comment by Seadragon91** [comment]
I used [CppCheck](http://cppcheck.sourceforge.net/)

**Comment by worktycho** [comment]
If your looking static analyzers, you might want to request access to the coverity scan system https://scan.coverity.com/projects/mc-server-mcserver

**Comment by Seadragon91** [comment]
Thank you, have requested access to it. Let's hunt bugs :)

**Pull request merged**